### PR TITLE
Fix governing body filter inital state

### PR DESCRIPTION
**Pull request by aliokan**
Related to BNB-555

### CODE REVIEW

**Comment by snenenenenenene** [comment]
I still found a bug when selecting multiple governing body classifications

**Comment by snenenenenenene** [comment]
i've worked a small solution myself if you want to take some inspiration for this fix [https://github.com/lblod/frontend-burgernabije-besluitendatabank/tree/fix/governing-body-classification-inserted](here)

File: app/components/filters/select-multiple-filter.ts [hunk 1]
```diff
@@ -33,22 +35,20 @@ export default class SelectMultipleFilterComponent extends FilterComponent<Signa
...
+    const haystack = await this.args.options;
+    const flattenedHaystack = haystack.reduce((acc, value) => {
+      if (Array.isArray(value.options)) {
+        return [...acc, ...value.options];
+      } else {
+        return [...acc, value as Option];
+      }
+    }, [] as Option[]);
 
     const results = deserializeArray(this.args.queryParam).flatMap(
```
Comment (snenenenenenene) [inline app/components/filters/select-multiple-filter.ts:53]: make it so the value gets split on ",". This way when multiple classifications get selected they'll get split up properly.

```ts
const results = deserializeArray(this.args.queryParam).flatMap(
      (queryParam) => {
        if (!queryParam) return [];
        const queryParamValue = this.getQueryParam(queryParam);
        const values = queryParamValue ? deserializeArray(queryParamValue) : [];
        return values
          .flatMap((value) => value.split(','))
          .map((value) => {
            return flattenedHaystack.find(
              (option) =>
                get(option, searchField) === value &&
                option['type'] === queryParam
            );
          })
          .filter(Boolean) as Option[];
      }
    );
```

**Review by snenenenenenene — CHANGES_REQUESTED** [review]

File: app/components/filters/select-multiple-filter.ts (hunk 1 above)
Comment (snenenenenenene) [reply app/components/filters/select-multiple-filter.ts:53]: Other than that it looks very good! :)

Comment (aliokan) [reply app/components/filters/select-multiple-filter.ts:53]: @snenenenenenene The problem was that we mix serialization separator. We used `+` on locations and `,` on governingBodyClassification. I refactor the `selectChange` method in order to apply `+` everywhere, that fix the bug too.

**Review by snenenenenenene — APPROVED** [review]
LGTM!

**Pull request merged**